FC-133 マッチング通知ダイアログを実装する

yukinkomob opened the pull request:
# ✨ What's done

- [x] ホーム画面表示時にマッチング通知ダイアログを表示
- [x] ホーム画面→トークルーム画面→チャットTOP画面に戻るの画面遷移ができるようにした。
- [x] `ChatService`クラスを作成し、チャット関連のデータは`ChatService`クラスに保持できるようにしました
- [x] `TabService`クラスを作成し、タブ画面の切り替えを他の画面にいてもできるように、`TabService`クラスに保持できるようにしました

# 🚩 What's not done

- [x] ルーティングを見直して広範囲に修正加えたのでコンフリクトすると思うので、
他の画面でコンフリクト修正必要になると思う。

# ✅ Test

- [x] iOS 14 Pro Maxエミュレータ、Pixel 4a API 33エミュレータにてダイアログが表示されることを確認
- [x] ダイアログ表示時にダイアログ外をタップするとダイアログが消去されることを確認

# 🔧 補足、備考

<img width="320" alt="Screen Shot 2022-11-26 at 2 03 42" src="https://user-images.githubusercontent.com/3205581/204030418-4e76c241-8d46-4cbb-88b9-51b7b898e373.png">

# 🔀 マージ条件

- [ ] レビューを通過する
- [ ] セルフマージする

yukinkomob committed:
ダイアログを表示（現状では文字列のみ）

yukinkomob committed:
Flutter Gen の利用設定

yukinkomob committed:
ダイアログに画像を表示

yukinkomob committed:
ダイアログに画像を表示

yukinkomob committed:
ダイアログの重複起動を抑止

yukinkomob committed:
ダイアログの重複起動を抑止

yukinkomob committed:
HomeViewをimmutableな形に戻し、クラス変数を利用してダイアログの有無を判定するように変更

yukinkomob committed:
HomeViewをimmutableな形に戻し、クラス変数を利用してダイアログの有無を判定するように変更

yukinkomob committed:
クラス変数の初期化に関するLINTチェックをignore

yukinkomob committed:
クラス変数の初期化に関するLINTチェックをignore

yukinkomob committed:
トーク画面への移動を追加

yukinkomob committed:
トーク画面への移動を追加

yukinkomob committed:
初期化方法を修正

yukinkomob committed:
初期化方法を修正

yukinkomob commented:
@king-kazu39 

お忙しいところすみません。トーク画面に移動しようとすると下記でアサーションが通知されております。
解決を試みたのですが、原因が分からず。。
コールスタックの確認、変数のウォッチなど、知りたい点があるので、お手すきの際にご指南いただけませんでしょうか。

![Screen Shot 2022-11-26 at 2 37 25](https://user-images.githubusercontent.com/3205581/204034250-2a464fe7-7fc1-4720-9be9-71363bdf168a.png)

yukinkomob committed:
毎回ダイアログを出すのは煩雑なので、1/3の確率で出す形に変更

yukinkomob committed:
毎回ダイアログを出すのは煩雑なので、1/3の確率で出す形に変更

king-kazu39 commented:
@yukinkomob 
ご対応ありがとうございます。
私が伝言できてなかったので、にっしーさんは悪くないんですが、
実はfluttergenで画像をメソッドに自動生成はしてるのでこちらのコミットはいらないかも。。
![スクリーンショット 2022-11-26 11 25 58](https://user-images.githubusercontent.com/25321380/204068876-cb855d27-3963-4760-b82d-a4b380a2c6f5.png)
なので、developプルした時にすぐ使える状態でした🙇
申し訳ないですがコミット取り消しの対応お願いしてもいいですか？

[fluttergenのドキュメント](https://github.com/Team-Ganju/favoritism_communication/blob/develop/docs/development-environment/flutter_gen.md)を読んでくれてるかわからないんですが、
今回はpubspec.yamlに書かない方法でfluttergen使おうと思ってるので、
homebrewかdart command-lineで有効にしておくようにお願いします🙇

あとsuggest出したいので、コミット取り消しが対応でき次第
ちょっと修正させてください🙏

king-kazu39 commented:
> @king-kazu39
> 
> お忙しいところすみません。トーク画面に移動しようとすると下記でアサーションが通知されております。 解決を試みたのですが、原因が分からず。。 コールスタックの確認、変数のウォッチなど、知りたい点があるので、お手すきの際にご指南いただけませんでしょうか。
> 
> ![Screen Shot 2022-11-26 at 2 37 25](https://user-images.githubusercontent.com/3205581/204034250-2a464fe7-7fc1-4720-9be9-71363bdf168a.png)

こちらに関してはそもそも画面遷移したいページが間違っていたので、suggestで対応しますね。

yukinkomob commented:
@king-kazu39 

> あとsuggest出したいので、コミット取り消しが対応でき次第
> ちょっと修正させてください🙏

flutter Gen の対応を削除しました。
suggest 対応をお願いできれば幸いです。

king-kazu39 commented:
@yukinkomob 
> @king-kazu39
> 
> > あとsuggest出したいので、コミット取り消しが対応でき次第
> > ちょっと修正させてください🙏
> 
> flutter Gen の対応を削除しました。 suggest 対応をお願いできれば幸いです。

ご対応ありがとうございます！
すいません。ソース見てて気づいたんですが、私の伝え方がかなりまずかったの気づきました。。
（色々誤解させてしまったかもしれないです。。申し訳ない）
suggest出す時に確認お願いします🙇

king-kazu39 committed:
ホーム画面からトークルームに遷移するために修正と整理

king-kazu39 committed:
不要なインポート削除

king-kazu39 committed:
Merge branch 'feature/FC-133-matching-notify-dialog' into feature/refactor-FC-133-matching-notify-dialog

king-kazu39 committed:
Revert "Flutter Gen の利用設定"

This reverts commit 77b01193f7e3af5c42332cae1e138231cd367eb6.

king-kazu39 committed:
♻️: ホーム画面からトークルーム画面に移動するときのルーティング修正とリファクタリング

# ✨ What's done

- [x] ホーム画面→トークルーム画面→チャットTOP画面に戻るの画面遷移ができるようにした。
- [x] `ChatService`クラスを作成し、チャット関連のデータは`ChatService`クラスに保持できるようにしました
- [x] `TabService`クラスを作成し、タブ画面の切り替えを他の画面にいてもできるように、`TabService`クラスに保持できるようにしました

# 🚩 What's not done

- [x] ルーティングを見直して広範囲に修正加えたのでコンフリクトすると思うので、
他の画面でコンフリクト修正必要になると思う。

# ✅ Test

- [x] AndroidエミュレータまたはiOSシミュレータで動作確認ずみ

yukinkomob commented:
下記で本件の修正がマージされたので、こちらはクローズする

♻️: ホーム画面からトークルーム画面に移動するときのルーティング修正とリファクタリング by king-kazu39 · Pull Request #37 · Team-Ganju/favoritism_communication
https://github.com/Team-Ganju/favoritism_communication/pull/37

king-kazu39 commented:
@yukinkomob 
コメントありがとうございます！
すいません、ややこしくて申し訳ないんですが、こちらのプルリクに自分のsuggestを取り込んでいるので
このプルリクをdevelopにマージしないとdevelopが最新にならないです。。
こちらもマージしちゃいますね。
本当にややこしくて申し訳ない🙇

yukinkomob commented:
@king-kazu39 
~~失礼しました。私が誤認してCloseしていたのでReopenしました。~~
Reopen操作が被ってしまったようで。。失礼しました。

king-kazu39 committed:
不要なインポート削除
